documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc

```diff
@@ -0,0 +1,82 @@
+[id='rest_v2_create_protobuf']
+= Creating Protobuf Schemas
+Create Protobuf schemas across {brandname} clusters with `POST` requests that
+include the content of a protobuf file in the payload.
+
+[source,options="nowrap",subs=attributes+]
+----
+POST /rest/v2/protobuf/{schemaName}
+----
+
+If the schema already exists, {brandname} returns `CONFLICT`. If the schema is
+not valid, {brandname} adds the schema and returns `BAD_REQUEST`.
```

Comment:
Not sure what "adds the schema" is here. Shouldn't it just return `BAD_REQUEST`?

Reply:
@karesti should we be more specific and say "adds the schema to the ___protobuf_metadata cache?

Reply:
No, because for me this cache is an implementation detail.
No because today even a wrong schema is stored. 
the thing is that if the file is not correct, a validation is done, and then you retrieve the validation result from a key linked to the file in the cache

Reply:
It being stored isn't an implementation detail, it is part of the interface of adding and verifying a schema. My comment is that the wording "adds the schema" seems a bit weird with it also returning `BAD_REQUEST`. If I was a user I would think that is a bit weird.

Even if the wording is "still registers the schema but returns a `BAD_REQUEST` response code. Or something like that.

Also I think we should add response code to these strings such as `CONFLICT`, because does that mean the body contains it or what. It is ambiguous.

Reply:
the bad_request error was a review of @gustavonalle 
I was responding to @oraNod suggestion of mentioning the cache name, because for me this is an implementation detail
for what it is better to say in english, you are far way better than me, so I'm ok about it

Reply:
Does it really adds the wrong schema to the cache?

Reply:
I think it adds the schema, oh maybe I understood it wrong ?

Reply:
My point is that when a user invokes this endpoint and they get a `BAD_REQUEST` response code back. I personally would never expect it to be inserted into the cache still. That is why I am saying we must be very explicit if this is the behavior we want and mention this as I would not expect it.

It ends up making it that the user may get `BAD_REQUEST` back now and then when they reinsert with the schema correct they would get `CONFLICT` which is not what I think anyone would expect.

Reply:
it stores the schema and adds an additional key

Reply:
I was never a fan of the whole inserting a schema puts an exception in another key/cache thing tbh. I feel like something like schema registration should be an operation that you either insert it and it is valid or it doesn't insert and you get an exception. If we mirror how the query cache does it, I am just saying it needs to be documented much more than "adds the schema" for such a case imo.

Reply:
@wburns I prefer created or ok for schemas with error too

Reply:
I will change that

Reply:
Yes, just checked and it does. Anyway, it does seem weird that "adds the schema" and return BAD_REQUEST at the same time. 

OTOH, I am not a particular fan of forcing the user to do two requests when registering the protobuf: one for the schema and another to check a key containing ```.error```. The REST endpoint should make things easier rather than simply exposing all the complexities behind it. The way @karesti implemented is fine to me  :+1: . 
I'd even go further by not mention in the documentation what it does, just say ```Registers the schema, return OK if registration is successful or BAD_REQUEST with the error otherwise```. 

Now the fact that it stores the wrong schema, require a tiny adjustment to the Resource: we can't return CONFLICT since when the schema is wrong, the user will not be able to override the schema with a POST again, it will already be there :disappointed: This sucks from the usability perspective, so we should only support PUT that always allow replacing.

Reply:
> it stores the schema and adds an additional key

When the schema is wrong, it stores in the cache but does not registers the schema in the serialization context. I don't think we should really worry about it. Just don't mention in the document about storing or caches with underscores in names, but document as a 'registration' operation: it either does or doesn't